src/Client.ts

```diff
@@ -237,7 +237,8 @@ export class Client extends EventEmitter<{
 
         processor
             .on("Disconnect", () => {
-                this.onDiscovered(host);
+                // The Connection class now handles reconnection internally
+                processor.log.warn("Processor disconnected, waiting for reconnection...");
```

Comment:
Logging here could be problematic here. I think an Reconnecting event should be added and bubbles up to the client.